### PR TITLE
Sy add expiration message

### PR DESCRIPTION
**Pull request by shahad-yousif**

### CODE REVIEW

File: config/locales/ar.yml [hunk 1]
```diff
@@ -100,6 +100,9 @@ ar:
       show:
     ad:
       new:
+        insufficient_business_balance: نفذ رصيد الكيدو المخصص لهذا العرض
```
Comment (ssharba) [inline config/locales/ar.yml]: remove duplicate translations please

File: app/serializers/customer_ad_serializer.rb [hunk 2]
```diff
@@ -35,10 +35,16 @@ def seen_ad
...
       I18n.translate('views.ad.new.ad_is_expired')
-    elsif object.ad.allocated_balance <= object.ad.transaction_ad_balance(object.ad.business_id)
+    elsif object.ad.allocated_balance <= object.ad.transaction_ad_balance(object.ad.business_id) && object.ad.is_deleted == false
       I18n.translate('views.ad.new.ad_run_out_of_tokens')
     elsif object.ad.is_paused
       I18n.translate('views.ad.new.ad_paused_by_business')
+    elsif object.ad.business.business_info.token_balance < object.ad.purchase_cost && object.ad.is_deleted == false
+      I18n.translate('views.ad.new.ad_run_out_of_tokens')
+    elsif object.ad.reserved_tokens.sum(:token_count) == object.ad.allocated_balance && object.ad.is_deleted == false
+      I18n.translate('views.ad.new.ad_run_out_of_tokens')
```
Comment (ssharba) [inline app/serializers/customer_ad_serializer.rb]: merge this two condition in one with OR

Comment (ssharba) [reply app/serializers/customer_ad_serializer.rb]: @MwMaki review please

Comment (MwMaki) [reply app/serializers/customer_ad_serializer.rb]: How about 
```
   # Ad passed the end date
    if object.ad.end_date < Date.today
      I18n.translate('views.ad.new.ad_is_expired')

    # Ad is paused 
    elsif object.ad.is_paused
      I18n.translate('views.ad.new.ad_paused_by_business')

    # Ad is deleted by the business
    elsif object.ad.is_deleted
      I18n.translate('views.ad.new.deleted_ad')
    # Ad OR business balance not enough 
    elsif object.ad.is_deleted == false
      trans_and_reservations = object.ad.transaction_ad_balance(object.ad.business_id) +  object.ad.reserved_tokens.sum(:token_count) 
      if (!ad.business.enough_balance?(ad.purchase_cost))  || (trans_and_reservations >= object.ad.allocated_balance)
        I18n.translate('views.ad.new.ad_run_out_of_tokens')
      end
    end
```

**Pull request merged**